community-operators/spark/sparkoperator.2.4.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,299 @@
+  maturity: alpha
+  version: 2.4.0
+  apiservicedefinitions: {}
+  provider:
+    name: GoogleCloudPlatform
+  links:
+  - name: GoogleCloudPlatform/spark-on-k8s-operator
+    url: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator
+  labels:
+    name: spark-operator
+  selector:
+    matchLabels:
+      name: spark-operator
+  customresourcedefinitions:
+    owned:
+    - kind: ScheduledSparkApplication
+      name: scheduledsparkapplications.sparkoperator.k8s.io
+      version: v1beta1
+      displayName: ScheduledSparkApplication
+      description: A configuration file for a ScheduledSparkApplication custom resource.
+    - kind: SparkApplication
+      name: sparkapplications.sparkoperator.k8s.io
+      version: v1beta1
+      displayName: SparkApplication
+      description: A configuration file for a SparkApplication custom resource.
+      resources:
+        - kind: Service
+          version: v1
+        - kind: Pod
+          version: v1
+      statusDescriptors:
+        - description: Current state of the application.
+          displayName: Application State
+          path: applicationState.state
+          x-descriptors: 
+            - 'urn:alm:descriptor:text'
+        - description: Indicates the error message in case of failures.
+          displayName: Error Message
+          path: applicationState.errorMessage
+          x-descriptors: 
+            - 'urn:alm:descriptor:io.kubernetes.phase:reason'
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: true
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    spec:
+      deployments:
+      - name: spark-operator
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              app.kubernetes.io/name: spark-operator
+              app.kubernetes.io/version: v2.4.0-v1beta1
+          strategy:
+            type: Recreate
+          template:
+            metadata:
+              creationTimestamp: null
+              initializers:
+                pending: []
+              labels:
+                app.kubernetes.io/name: spark-operator
+                app.kubernetes.io/version: v2.4.0-v1beta1
+            spec:
+              containers:
+              - args:
+                - -logtostderr
+                - -install-crds=false
+                - -namespace=$(WATCH_NAMESPACE)
+                image: gcr.io/spark-operator/spark-operator:v2.4.0-v1beta1-latest
+                imagePullPolicy: Always
+                name: spark-operator
+                resources: {}
+                env:
+                - name: WATCH_NAMESPACE
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.annotations['olm.targetNamespaces']
+              serviceAccountName: spark-operator
+      permissions:
+      - serviceAccountName: spark
```

Comment:
Is this serviceAccount used by the spark application?

Reply:
Yes. This service account is meant for the driver/executor pods that the spark-operator will create for each CR. I've added a note about this in the `Before you start` section of the CSV description as well.
https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/quick-start-guide.md#about-the-service-account-for-driver-pods

If OLM doesn't create it for them, then users would need to create a service account themselves and add it to the `anyuid` SCC in order to run their spark applications.
```console
$ oc adm policy add-scc-to-user anyuid -z spark -n <namespace>
```
I'm not sure if all users would have the required permissions to do the above so I thought it's easier if OLM does it for them.

Reply:
This is good, other operators follow a similar workflow.

Reply:
`anyuid` is not required, Spark images have the required logic to run with the specific user available.

Reply:
@skonto If this is true, you are welcome to open a PR updating the spark-gcp operator

Reply:
thanks @SamiSousa